tasks/resolve_reference.rb

```diff
@@ -40,14 +41,29 @@ def resolve_reference(opts)
     raise "Error occured binding to the domain controller #{ldap_properties[:host]}: #{ldap.get_operation_result.message} #{ldap.get_operation_result.error_message}" unless ldap.get_operation_result.code.zero?
 
     # Execute the search
+    filter_dns_hostnames = opts[:filter_dns_hostnames]
+    filter_older_attribute = opts[:filter_older_attribute]
+    filter_older_than_days = opts[:filter_older_than_days]
+    ad_attributes = DEFAULT_AD_ATTRIBUTES.dup
+    if filter_older_attribute && filter_older_than_days
+      filter_older = true
+      filter_older_attribute_s = filter_older_attribute.downcase.to_s
+      ad_attributes << filter_older_attribute
+    end
     calc_transport = opts[:calculate_transport] || true
     result = []
     ldap.search(
       base:         x500_domain,
       filter:       Net::LDAP::Filter.eq( 'objectCategory', 'computer' ),
-      attributes:   DEFAULT_AD_ATTRIBUTES,
+      attributes:   ad_attributes,
       return_result: false
     ) do |entry|
+      if filter_dns_hostnames
```

Comment:
Personally, I'd do the filtering at the LDAP search query level, instead of retrieving ALL AD objects and then doing client side filtering (https://github.com/glennsarti/bolt_ad_inventory/blob/master/tasks/resolve_reference.rb#L47)

It's much faster for the both the client and server

Reply:
Fixed.